### DISCUSSION TITLE
Custom IdRepo

### DISCUSSION THREAD

**Discussion by meha-patel2020** [OpenIdentityPlatform/OpenAM#611] [Q&A] [answered]
Hi,
I am trying to create a Custom Id Repo which uses MongoDB as a repository.
WIKI used as reference: https://wikis.forgerock.org/confluence/display/openam/Develop+and+Deploy+a+Custom+IdRepo+Plugin

I was successfully able to add a subschema while selecting a Data Store. 
![image](https://github.com/OpenIdentityPlatform/OpenAM/assets/78463052/3083838c-5413-4626-8932-b6368a67426d)

After configuring the mongodb(custom) datastore when I move to the **Subjects** tab I get a message "**There are no data stores configured for this realm.**" even though a data store is configured.
![image](https://github.com/OpenIdentityPlatform/OpenAM/assets/78463052/cf66f798-0302-4b99-843a-ee45b347f293)

Not sure what exactly is missing.
Any help will be much appreciated.

**Comment by maximthomas** [comment, marked as answer]
@meha-patel2020 seems, you have not implemented functions from IdRepo class
```java
public Set<IdType> getSupportedTypes() 
```
So default implementation returns and empty set so OpenAM thinks your repository does not support Users and Groups identity types and shows the error message.
 As an example you can check Apache Cassandra implementation [Repo.java](https://github.com/OpenIdentityPlatform/OpenAM/blob/master/openam-cassandra/openam-cassandra-datastore/src/main/java/org/openidentityplatform/openam/cassandra/Repo.java)
